hooks.go

```diff
@@ -24,11 +28,13 @@ func (hooks LevelHooks) Add(hook Hook) {
 // Fire all the hooks for the passed level. Used by `entry.log` to fire
 // appropriate hooks for a log entry.
 func (hooks LevelHooks) Fire(level Level, entry *Entry) error {
+	var aggErr error
+
 	for _, hook := range hooks[level] {
 		if err := hook.Fire(entry); err != nil {
-			return err
+			aggErr = fmt.Errorf("%s | %s", aggErr, err)
```

Comment:
There aren't great solutions for this without getting fancy. You could fold it into a lightweight interface that does the formatting and collection management.

Where is this called? It might be reasonable just to log these failures in place.